shell/browser/api/electron_api_web_contents.cc

```diff
@@ -362,14 +363,64 @@ struct Converter<scoped_refptr<content::DevToolsAgentHost>> {
 
 template <>
 struct Converter<content::NavigationEntry*> {
+  static bool FromV8(v8::Isolate* isolate,
+                     v8::Local<v8::Value> val,
+                     content::NavigationEntry** out) {
+    gin_helper::Dictionary dict;
+    if (!gin::ConvertFromV8(isolate, val, &dict))
+      return false;
+
+    std::string url_str;
+    std::string title;
+    std::string encoded_page_state;
+
+    if (!dict.Get("url", &url_str) || !dict.Get("title", &title))
+      return false;
+
+    GURL url(url_str);
```

Comment:
Pretty sure we have a `GURL` converter already somewhere, saves the duplication if we use that